include/arch/arm/arch/machine/hardware.h

```diff
@@ -30,3 +30,11 @@ typedef word_t vm_fault_type_t;
 #define L1_CACHE_LINE_SIZE_BITS CONFIG_L1_CACHE_LINE_SIZE_BITS
 #define L1_CACHE_LINE_SIZE BIT(L1_CACHE_LINE_SIZE_BITS)
 
+/* Used to align the big kernel lock to the exclusive reservation granule size.
+ * Without this nearby writes can delay atomic operations implemented with looping
+ * exclusive load/store instructions for an undefined time.
+ *
+ * Usually equal to L1_CACHE_LINE_SIZE, but 2k is the maximum for SMP systems.
```

Comment:
Can you refer to any source here, is this ARMv8 or ARMv7 also?

Reply:
Good idea, done.